src/query/builder.js

```diff
@@ -53,6 +53,9 @@ assign(Builder.prototype, {
     if (!isUndefined(this._options)) {
       cloned._options = clone(this._options);
     }
+    if (!isUndefined(this._hookContext)) {
+      cloned._hookContext = clone(this._hookContext);
```

Comment:
@elhigu does it make sense to clone the context object as well? I imagine someone would want the same context object to be passed around (to be able to build up on it) but I just added added this to conform with the rest

Reply:
well at least.. you really cannot do deep clone for it, so I would say that it is enough to copy only root element and if someone like some how shared content they just need to add nested structure to context and inner content will be just references to the same data... not sure if that is understandable at all..

Reply:
I understand.. but is there a reason to clone the root context itself?

Reply:
@joelmukuthu yes. to be able to have different contexts for cloned queries. So I can clone query and then change / merge its root context without modifying the original. So that way it be more versatile and user can decide how to use it.

Reply:
Though it will be also caveat that must be documented.

Reply:
Makes perfect sense, thanks for the explanation!